src/main/java/ro/utcn/sd/boti/stackoverflow/service/StackoverflowService.java

```diff
@@ -57,14 +59,45 @@ public void removeQuestion(int id) {
         return repository.findByText(s);
     }
 
+    @Transactional
     public List<Tag> listTags() { return repositoryFactory.getTagRepository().findAll(); }
 
+    @Transactional
     public List<Question> searchByTag(int id) {
         Tag tag = repositoryFactory.getTagRepository().findById(id).get();
         return tag.getQuestions();
     }
 
+    @Transactional
     public Optional<User> findByUserName(String username) {
         return repositoryFactory.getUserRepository().findByUserName(username);
     }
+
+    @Transactional
+    public Optional<Question> findQuestionById(int id) {
+        return repositoryFactory.getQuestionRepository().findById(id);
+    }
+
+    @Transactional
+    public Answer addAnswer(User user, Question question, String text) {
+        Answer answer = new Answer(user, question, text, new SimpleDateFormat("yyyy-MM-dd HH:mm:ss").format(new Date()));
+        question.getAnswers().add(answer);
+        return repositoryFactory.getAnswerRepository().save(answer);
+    }
+
+    @Transactional
+    public Optional<Answer> findAnswerById(int id){
```

Comment:
You only have one service class. You should split it up (either Table Module = one service per each business object; e.g. an AnswerService and a QuestionService or one service per business process; e.g. AskQuestionService, AnswerQuestionService, etc).